### ISSUE TITLE
Type parameter works directly, but fails inside `List<>`

### ISSUE THREAD

**Issue by andy-ms** [pythonnet/pythonnet#962]
### Environment

-   Pythonnet version: 2.4.0
-   Python version: Python 3.7.4
-   Operating System: Windows 10 (18362.356)

### Details

-   Describe what you were trying to get done.

Call a generic method taking a list parameter.

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

C# library:

```cs
using System.Collections.Generic;

namespace CSLib
{
    public static class CSLib
    {
        public static bool GetBool() =>
            true;

        public static void TestValue<T>(T t) {}

        public static List<bool> GetBoolList() =>
            new List<bool>() { true, false };

        public static void TestList<T>(List<T> list) {}

        public static void Test()
        {
            TestValue<bool>(GetBool());
            TestList<bool>(GetBoolList());
        }
    }
}
```

Python code:

```python
from clr import AddReference
from pathlib import Path

AddReference(str(Path.cwd() / "cslib/bin/Debug/netstandard2.0/cslib.dll"))

from System import Boolean
from CSLib import CSLib

CSLib.TestValue[Boolean](CSLib.GetBool())
CSLib.TestList[Boolean](CSLib.GetBoolList())
```

It works fine when the type argument is used directly, but fails when it is inside `List<>`.

-   If there was a crash, please include the traceback here.

```python
Traceback (most recent call last):
  File "a.py", line 11, in <module>
    CSLib.TestList[Boolean](CSLib.GetBoolList())
TypeError: No method matches given arguments for TestList
```

**Comment by filmor** [comment]
This is due to the implicit conversion of .NET lists to Python lists again. Since we don't do it the other way round (and it would also be very tricky to get that right), when you get `CSLib.GetBoolList()` you end up with a Python list `[True]` which doesn't match the method signature.

I'm working on removing the implicit conversion in #963.

**Comment by lostmsu** [comment]
This should now be fixed in 3.0
